marshmallow/fields.py

```diff
@@ -616,6 +617,118 @@ def _deserialize(self, value, attr, data, **kwargs):
+        `typing.NamedTuple`, using a nested schema for them is more appropriate
+        than using a `Tuple` field.
+
+    :param Iterable[Field] tuple_fields: An iterable of field classes or
+        instances.
+    :param kwargs: The same keyword arguments that :class:`Field` receives.
+    """
+
+    default_error_messages = {
+        'invalid': 'Not a valid tuple.',
+    }
+
+    def __init__(self, tuple_fields, *args, **kwargs):
+        super(Tuple, self).__init__(*args, **kwargs)
+        if not utils.is_collection(tuple_fields):
+            raise ValueError(
+                'tuple_fields must be an iterable of Field classes or '
+                'instances.',
+            )
+        self.tuple_fields = [
+            self._get_instance_from_field_class_or_instance(cls_or_instance)
+            for cls_or_instance in tuple_fields
+        ]
+        self.validate_length = Length(equal=len(self.tuple_fields))
+
+    def _bind_to_schema(self, field_name, schema):
+        super(Tuple, self)._bind_to_schema(field_name, schema)
+        new_tuple_fields = []
+        for container in self.tuple_fields:
+            new_container = copy.deepcopy(container)
+            new_container.parent = self
+            new_container.name = field_name
+            new_tuple_fields.append(new_container)
+        self.tuple_fields = new_tuple_fields
+
+    @staticmethod
+    def _get_instance_from_field_class_or_instance(cls_or_instance):
+        if isinstance(cls_or_instance, type):
+            if not issubclass(cls_or_instance, FieldABC):
+                raise ValueError(
+                    'Elements of "tuple_fields" must be subclasses or '
+                    'instances of marshmallow.base.FieldABC.',
+                )
+            return cls_or_instance()
+        else:
+            if not isinstance(cls_or_instance, FieldABC):
+                raise ValueError(
+                    'Elements of "tuple_fields" must be subclasses or '
+                    'instances of marshmallow.base.FieldABC.',
+                )
+            return cls_or_instance
+
+    def get_value(self, obj, attr, accessor=None):
+        """Return the value for a given key from an object."""
+        value = super(Tuple, self).get_value(obj, attr, accessor=accessor)
+        if any(container.attribute for container in self.tuple_fields):
+            if not utils.is_collection(value):
+                self.fail('invalid')
+            result = []
+            for container, each in zip(self.tuple_fields, value):
+                if container.attribute is None:
+                    result.append(each)
+                else:
+                    result.append(
+                        container.get_value(each, container.attribute),
+                    )
+            return tuple(result)
+        return value
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if value is None:
+            return None
+        if not utils.is_collection(value):
```

Comment:
Validation should not happen on serialization, for performance reasons. `value` is already assumed to be valid.